smart-metastore/src/main/java/org/smartdata/metastore/dao/FileStateDao.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.metastore.dao;
+
+import org.smartdata.model.FileState;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import javax.sql.DataSource;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FileStateDao {
+  private DataSource dataSource;
+
+  public void setDataSource(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  public FileStateDao(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  public void insertUpate(FileState fileState) {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    String sql = "REPLACE INTO file_state (path, type, stage) VALUES (?,?,?);";
+    jdbcTemplate.update(sql, fileState.getPath(), fileState.getFileType().getValue(),
+        fileState.getFileStage().getValue());
+  }
+
+  public FileState getByPath(String path) {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    return jdbcTemplate.queryForObject("SELECT * FROM file_state WHERE path = ?",
```

Comment:
"SELECT * FROM file_state WHERE path = ?". Make "file_state" a property ([ActionDao](https://github.com/Intel-bigdata/SSM/blob/trunk/smart-metastore/src/main/java/org/smartdata/metastore/dao/ActionDao.java#L40)).

Reply:
OK. It is done now.